src/humanloop/client.py

```diff
@@ -78,11 +107,304 @@ def __init__(
             follow_redirects=follow_redirects,
             httpx_client=httpx_client,
         )
+
+        if opentelemetry_tracer_provider is not None:
+            self._tracer_provider = opentelemetry_tracer_provider
+        else:
+            self._tracer_provider = TracerProvider(
+                resource=Resource(
+                    attributes={
+                        "instrumentor": "humanloop.sdk",
+                    }
+                ),
+            )
+        instrument_provider(provider=self._tracer_provider)
+        self._tracer_provider.add_span_processor(
+            HumanloopSpanProcessor(
+                exporter=HumanloopSpanExporter(
+                    client=self,
+                )
+            ),
+        )
+
+        if opentelemetry_tracer is not None:
+            set_humanloop_sdk_tracer(opentelemetry_tracer)
+        else:
+            tracer = self._tracer_provider.get_tracer("humanloop.sdk")
+            set_humanloop_sdk_tracer(tracer)
+
         eval_client = ExtendedEvalsClient(client_wrapper=self._client_wrapper)
         eval_client.client = self
         self.evaluations = eval_client
         self.prompts = ExtendedPromptsClient(client_wrapper=self._client_wrapper)
 
+    def prompt(
+        self,
+        *,
+        path: Optional[str] = None,
+        model: Optional[str] = None,
+        endpoint: Optional[ModelEndpoints] = None,
+        template: Optional[PromptKernelRequestTemplate] = None,
+        provider: Optional[ModelProviders] = None,
+        max_tokens: Optional[int] = None,
+        temperature: Optional[float] = None,
+        top_p: Optional[float] = None,
+        stop: Optional[PromptKernelRequestStop] = None,
+        presence_penalty: Optional[float] = None,
+        frequency_penalty: Optional[float] = None,
+        other: Optional[dict[str, Optional[Any]]] = None,
+        seed: Optional[int] = None,
+        response_format: Optional[ResponseFormat] = None,
+    ):
+        """Decorator for declaring a (Prompt)[https://humanloop.com/docs/explanation/prompts] in code.
+
+        The decorator intercepts calls to LLM provider APIs and creates
+        a new Prompt file based on the hyperparameters used in the call.
+        If a hyperparameter is specified in the `@prompt` decorator, then
+        they override any value intercepted from the LLM provider call.
+
+        If the (Prompt)[https://humanloop.com/docs/explanation/prompts] already exists
+        on the specified path, a new version will be upserted when any of the above change.
+
+        Here's an example of declaring a (Prompt)[https://humanloop.com/docs/explanation/prompts] in code:
+
+        ```python
+        @prompt(template="You are an assistant on the following topics: {topics}.")
```

Comment:
Should update our template examples and docstrings to use a chat template format. List of messages that have {{}} vars in their content.

This is the more likely use case.

The type of template should be the same union type in the SDK

Reply:
Still outstanding i think?